arctic/store/metadata_store.py

```diff
@@ -57,8 +59,50 @@ def __repr__(self):
         return str(self)
 
     @mongo_retry
-    def list_symbols(self):
-        return self.distinct('symbol')
+    def list_symbols(self, regex=None, as_of=None, **kwargs):
+        """
+         Return the symbols in this library.
+
+         Parameters
+         ----------
+         as_of : `datetime.datetime`
+            filter symbols valid at given time
+         regex : `str`
+             filter symbols by the passed in regular expression
+         kwargs :
+             kwarg keys are used as fields to query for symbols with metadata matching
+             the kwargs query
```

Comment:
Maybe add documentation saying that this operation will scan over all documents in the collection if any of these parameters are passed.

I understand that the operation self.distinct(symbol) is not that fast either, but at least it should be a covered query. Therefore, hopefully, using only the index and not loading every document into memory.

Reply:
I've updated the aggregation pipeline to avoid full scan as much as possible.